src/MongoDB.Driver/Search/SearchDefinitions.cs

```diff
@@ -0,0 +1,379 @@
+﻿// Copyright 2010-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.GeoJsonObjectModel;
+
+namespace MongoDB.Driver.Search
+{
+    internal sealed class AutocompleteSearchDefinition<TDocument> : OperatorSearchDefinition<TDocument>
+    {
+        private readonly QueryDefinition _query;
+        private readonly AutocompleteTokenOrder _tokenOrder;
+        private readonly FuzzyOptions _fuzzy;
+
+        public AutocompleteSearchDefinition(
+            QueryDefinition query,
+            PathDefinition<TDocument> path,
+            AutocompleteTokenOrder tokenOrder,
+            FuzzyOptions fuzzy,
+            ScoreDefinition<TDocument> score)
+            : base(OperatorType.Autocomplete, path, score)
+        {
+            _query = Ensure.IsNotNull(query, nameof(query));
+            _tokenOrder = tokenOrder;
+            _fuzzy = fuzzy;
+        }
+
+        private protected override BsonDocument RenderOperator(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry) =>
+           new()
+           {
+                { "query", _query.Render() },
+                { "tokenOrder", _tokenOrder.ToCamelCase(), _tokenOrder == AutocompleteTokenOrder.Sequential },
```

Comment:
```suggestion
                { "tokenOrder", _tokenOrder.ToCamelCase(), _tokenOrder != AutocompleteTokenOrder.Any },
```

This should be more future-proof in case they add other token order options.

Reply:
Thanks, good catch!